Ctor 457 new pack network juniper mseries netconf (#4416)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/network-routers-juniper-mseries-netconf.md

---
id: network-routers-juniper-mseries-netconf
title: Juniper M-Series Netconf
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Juniper M-Series Netconf** connector through the

> **Configuration > Monitoring Connectors** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

> **Configuration > Monitoring Connectors** menu:

Reviewer 2: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

| service-rpm-probe-nrtj-delay-jitter          | N/A   |
| service-rpm-probe-nrtj-delay-stdev           | N/A   |

</TabItem>
<TabItem value="RSVP-sessions" label="RSVP-sessions">

| Name                                           | Unit  |
|:-----------------------------------------------|:------|
| rsvp.sessions.detected.count                   | count |
| status                                         | N/A   |
| *rsvp*#rsvp.session.lsp.traffic.bytespersecond | B/s   |

</TabItem>
</Tabs>

## Prerequisites

The Netconf service must be enabled on the Juniper M-Series router, and the router must be configured to allow connections from the Centreon server or poller.
A user account with the necessary rights to access Netconf data must also be created.

## Installing the monitoring connector

### Pack

The installation procedures for monitoring connectors are slightly different depending on [whether your license is offline or online](../getting-started/how-to-guides/connectors-licenses.md).

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connectors** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

> **Configuration > Monitoring Connectors** menu.

Reviewer 2: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-network-routers-juniper-mseries-netconf
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-network-routers-juniper-mseries-netconf
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-network-routers-juniper-mseries-netconf
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-network-routers-juniper-mseries-netconf
```

</TabItem>
</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Juniper M-Series Netconf** connector through

> the **Configuration > Monitoring Connectors** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

> the **Configuration > Monitoring Connectors** menu.

Reviewer 2: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

| CRITICALSERVICERPMPROBERTTDELAYSTDEV    | Define the service RPM probe route-trip time delay standard deviation conditions to match for the status to be CRITICAL                                                                                             |               |           |
| WARNINGSERVICESDETECTED                 | Define the services detected conditions to match for the status to be WARNING                                                                                                                                       |               |           |
| CRITICALSERVICESDETECTED                | Define the services detected conditions to match for the status to be CRITICAL                                                                                                                                      |               |           |
| WARNINGSTATUS                           | Define the conditions to match for the status to be WARNING. You can use the following variables: C\<%\{testName\}\>, C\<%\{probeType\}\>, C\<%\{sourceAddress\}\>, C\<%\{targetAddress\}\>, C\<%\{probeStatus\}\>  |               |           |

> | CRITICALSTATUS                          | Define the conditions to match for the status to be CRITICAL. You can use the following variables: C\<%\{testName\}\>, C\<%\{probeType\}\>, C\<%\{sourceAddress\}\>, C\<%\{targetAddress\}\>, C\<%\{probeStatus\}\> |               |           |

Reviewer: C?

> | CRITICALSTATUS                          | Define the conditions to match for the status to be CRITICAL. You can use the following variables: C\<%\{testName\}\>, C\<%\{probeType\}\>, C\<%\{sourceAddress\}\>, C\<%\{targetAddress\}\>, C\<%\{probeStatus\}\> |               |           |

Reviewer 2: ```suggestion
| WARNINGSTATUS                           | Define the conditions to match for the status to be WARNING. You can use the following variables: <%\{testName\}\>, <%\{probeType\}\>, <%\{sourceAddress\}\>, <%\{targetAddress\}\>, <%\{probeStatus\}\>  |               |           |
| CRITICALSTATUS                          | Define the conditions to match for the status to be CRITICAL. You can use the following variables: <%\{testName\}\>, <%\{probeType\}\>, <%\{sourceAddress\}\>, <%\{targetAddress\}\>, <%\{probeStatus\}\> |               |           |
```

> | CRITICALSTATUS                          | Define the conditions to match for the status to be CRITICAL. You can use the following variables: <%\{testName\}\>, <%\{probeType\}\>, <%\{sourceAddress\}\>, <%\{targetAddress\}\>, <%\{probeStatus\}\> |               |           |

Reviewer 2: ```suggestion
| WARNINGSTATUS                           | Define the conditions to match for the status to be WARNING. You can use the following variables: %\{testName\}, %\{probeType\}, %\{sourceAddress\}, %\{targetAddress\}, %\{probeStatus\}  |               |           |
| CRITICALSTATUS                          | Define the conditions to match for the status to be CRITICAL. You can use the following variables: %\{testName\}, %\{probeType\}, %\{sourceAddress\}, %\{targetAddress\}, %\{probeStatus\} |               |           |
```

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
</Tabs>

### Discovery rules

#### Service discovery

| Rule name                                | Description                                                   |
|:-----------------------------------------|:--------------------------------------------------------------|
| Net-Juniper-Mseries-Netconf-Bgp          |                                                               |
| Net-Juniper-Mseries-Netconf-Disks        | Discover the disk partitions and monitor space occupation     |
| Net-Juniper-Mseries-Netconf-Interfaces   | Discover network interfaces and monitor bandwidth utilization |
| Net-Juniper-Mseries-Netconf-Ldp          |                                                               |
| Net-Juniper-Mseries-Netconf-Lsp          |                                                               |
| Net-Juniper-Mseries-Netconf-Rsvp         |                                                               |

> | Net-Juniper-Mseries-Netconf-Services-Rpm | Discover services and monitor their system usage              |

Reviewer 2: ```suggestion
| Net-Juniper-Mseries-Netconf-Bgp               | Discover BGP peers and monitor them                                        |
| Net-Juniper-Mseries-Netconf-Disks             | Discover the disk partitions and monitor space occupation        |
| Net-Juniper-Mseries-Netconf-Interfaces      | Discover network interfaces and monitor bandwidth utilization |
| Net-Juniper-Mseries-Netconf-Ldp                | Discover LDP sessions and monitor their status                          |
| Net-Juniper-Mseries-Netconf-Lsp                 | Discover LSP and monitor their status                                        |
| Net-Juniper-Mseries-Netconf-Rsvp               | Discover RSVP sessions and monitor them                                 |
| Net-Juniper-Mseries-Netconf-Services-Rpm | Discover RPM services and monitor them                                   |
```

| status                                       | N/A   |
| *services*~service.rpm.probe.loss.percentage | %     |
| service-rpm-probe-rtt-delay-average          | N/A   |
| service-rpm-probe-rtt-delay-jitter           | N/A   |
| service-rpm-probe-rtt-delay-stdev            | N/A   |
| service-rpm-probe-prtj-delay-average         | N/A   |
| service-rpm-probe-prtj-delay-jitter          | N/A   |
| service-rpm-probe-prtj-delay-stdev           | N/A   |
| service-rpm-probe-nrtj-delay-average         | N/A   |
| service-rpm-probe-nrtj-delay-jitter          | N/A   |
| service-rpm-probe-nrtj-delay-stdev           | N/A   |

</TabItem>
<TabItem value="RSVP-sessions" label="RSVP-sessions">

| Name                                           | Unit  |
|:-----------------------------------------------|:------|
| rsvp.sessions.detected.count                   | count |
| status                                         | N/A   |
| *rsvp*#rsvp.session.lsp.traffic.bytespersecond | B/s   |

</TabItem>
</Tabs>

## Prerequisites

*Specify prerequisites that are relevant. You may want to just provide a link\n\
to the manufacturer official documentation BUT you should try to be as complete\n\

> as possible here as it will save time to everybody.*

Reviewer 2: Il faudra ajouter les pré-requis

| WARNINGBGPPEERDETECTED           | Warning threshold for number of BGP peers detected                                                                                                                                                                        |                                  |           |
| CRITICALBGPPEERDETECTED          | Critical threshold for number of BGP peers detected                                                                                                                                                                       |                                  |           |
| WARNINGBGPPEERRIBPREFIXESACTIVE  |                                                                                                                                                                                                                           |                                  |           |

> | CRITICALBGPPEERRIBPREFIXESACTIVE |                                                                                                                                                                                                                           |                                  |           |

Reviewer 2: ```suggestion
| WARNINGBGPPEERRIBPREFIXESACTIVE  | Threshold                                                                                                                                                                                                                          |                                  |           |
| CRITICALBGPPEERRIBPREFIXESACTIVE | Threshold                                                                                                                                                                                                                          |                                  |           |
```

</TabItem>
<TabItem value="OSPF" label="OSPF">

| Macro                     | Description                                                                                                                                                                        | Default value         | Mandatory |
|:--------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------|:---------:|
| FILTERNEIGHBORADDRESS     | Filter neighbors by address (can be a regexp)                                                                                                                                      |                       |           |
| WARNINGNEIGHBORSCHANGED   |                                                                                                                                                                                    |                       |           |

> | CRITICALNEIGHBORSCHANGED  |                                                                                                                                                                                    |                       |           |

Reviewer 2: ```suggestion
| WARNINGNEIGHBORSCHANGED   | Threshold                                                                                                                                                                                   |                       |           |
| CRITICALNEIGHBORSCHANGED  | Threshold                                                                                                                                                                                   |                       |           |
```
